doc/source/ray-core/scheduling/ray-oom-prevention.rst

```diff
@@ -16,117 +16,92 @@ What is the memory monitor?
+- ``RAY_memory_monitor_refresh_ms (int, defaults to 250)`` is the interval to check memory usage and kill tasks or actors if needed. Task killing is disabled when this value is 0. The memory monitor selects and kills one task at a time and waits for it to be killed before choosing another one, regardless of how frequent the memory monitor runs.
 
 - ``RAY_memory_usage_threshold (float, defaults to 0.95)`` is the threshold when the node is beyond the memory
-  capacity. If the memory usage is above this value and the free space is
-  below min_memory_free_bytes then it will start killing processes to free up space. Ranges from [0, 1].
+  capacity. If the memory usage is above this fraction it will start killing processes to free up memory. Ranges from [0, 1].
 
-- ``RAY_task_oom_retries (int, defaults to 15):`` The number of retries for the task or actor when
-  it fails due to the process being killed by the memory monitor.
-  If the task or actor is not retriable then this value is not used. This value is used
-  only when the process is killed by the memory monitor, and the retry counter of the
-  task or actor (:ref:`max_retries <task-fault-tolerance>` or :ref:`max_restarts <actor-fault-tolerance>`) is used when it fails 
-  in other ways. If the process is killed by the operating system OOM killer it will use the task retry and not ``task_oom_retries``.
-  It will retry indefinitely if this value is set to -1.
-
-  When the task is retried due to OOM it applies a delay before re-executing the task. The delay is calculated as
-
-  .. code-block:: bash
+Using the Memory Monitor
+------------------------
 
-      delay_seconds = 2 ^ attempt
+Retry policy
+~~~~~~~~~~~~
 
-  Where the first retry will be delayed by 1 second as ``attempt`` starts from 0.
+When a task or actor is killed by the memory monitor it will be retried with exponential backoff. There is a cap on the retry delay, which is 60 seconds. There is no limit to the number of retries when the task is killed by the memory monitor. For other types of failures (e.g., node failure), it uses :ref:`max_retries <task-fault-tolerance>`, or :ref:`max_restarts <actor-fault-tolerance>` and :ref:`max_task_retries <actor-fault-tolerance>` depending on whether it is a task or actor.
 
-  .. note::
+Worker killing policy
+~~~~~~~~~~~~~~~~~~~~~
 
-      Keep the value of ``RAY_task_oom_retries`` low, below 25, to avoid extremely long delays as it is using exponential backoff.
-      The delay is capped at 1 minute.
-  
+The memory monitor will ensure liveness by having at least one task or actor execute for each caller. If the last task of a caller is killed then it will fail the workload. Refer to :ref:`how to address memory issues <addressing-memory-issues>` on how to adjust the workload to make it pass when this happens. Refer to the :ref:`code example below <last-task-example>` for how this works.
```

Comment:
It doesn't apply to actors right?

Reply:
It might, if max_restarts > 0

I think i put a note in another section to clarify actors won't retry by default, so the way we implemented non-retriable actors belonging to a separate group could be treated as implementation detail?

Reply:
By default , it won't, so we have a note below this paragraph:

But let me know if it doesn't make sense

![Screen Shot 2023-02-08 at 4 40 50 PM](https://user-images.githubusercontent.com/22830327/217684928-5ab675fb-8fa4-42b7-a535-ba792341788c.png)

Reply:
It's a bit ambiguous reading, how about this rewording instead?

```
The memory monitor avoids infinite loops of task retries by ensuring at least one task is able to run for each caller on each node. If it is unable to ensure this, the task will be failed. Note that this is only an issue for tasks, since the memory monitor will not indefinitely retry actors.

```